src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js

```diff
@@ -88,10 +108,20 @@ export default ajaxCaller(class LaunchAnalysisModal extends Component {
         this.createSetAndLaunch(entities)
       }
     } else if (type === EntitySelectionType.processFromSet) {
-      const { entityType, name } = selectedEntities
-      this.launch(entityType, name, `this.${rootEntityType}s`)
+      if (_.size(selectedEntities) === 1) {
+        const { entityType, name } = selectedEntities[0]
+        this.launch(entityType, name, `this.${rootEntityType}s`)
+      } else {
+        const entities = _.reduce((acc, { attributes: { [`${rootEntityType}s`]: { items } } }) => _.concat(acc, _.map('entityName', items)), [],
+          selectedEntities)
+        this.createSetAndLaunch(entities)
+      }
     } else if (type === EntitySelectionType.chooseSet) {
-      this.launch(rootEntityType, selectedEntities['name'])
+      if (_.size(selectedEntities) === 1) {
+        this.launch(rootEntityType, selectedEntities['name'])
+      } else {
+        this.launchParallel()
```

Comment:
TOL: Going forward we could consider making multi-launch the general case, with single-launch as a specific case of that. That would allow us to leverage the nice status and error handling for both.